src/chatkit.ts

```diff
@@ -292,137 +354,137 @@ export default class Chatkit {
 
   // Authorizer interactions
 
-  createRoomRole(name: string, permissions: Array<string>): Promise<void> {
-    return this.createRole(name, 'room', permissions);
+  createRoomRole(options: CreateRoleOptions): Promise<void> {
+    return this.createRole({
+      ...options,
+      scope: 'room',
+    });
   }
 
-  createGlobalRole(name: string, permissions: Array<string>): Promise<void> {
-    return this.createRole(name, 'global', permissions);
+  createGlobalRole(options: CreateRoleOptions): Promise<void> {
+    return this.createRole({
+      ...options,
+      scope: 'global',
+    });
   }
 
-  private createRole(name: string, scope: string, permissions: Array<string>): Promise<void> {
+  private createRole(options: CreateScopedRoleOptions): Promise<void> {
     return this.authorizerInstance.request({
       method: 'POST',
       path: `/roles`,
       headers: {
         'Content-Type': 'application/json'
       },
-      body: { scope, name, permissions },
+      body: options,
```

Comment:
might be worth being explicit here, since in the javascript you'll lose all type safety and suddenly you're passing an arbitrary value as the body. Then again, there's no other type checking anywhere, so again, this might be out of scope.